code/datums/diseases/advance/advance.dm

```diff
@@ -385,7 +400,7 @@
 				if(!D.HasSymptom(S))
 					D.symptoms += S
 					i -= 1
-	while(i > 0)

```

Comment:
You are making it worse optimization-wise
Id say you can make it faster just checking it to not be 0 but its byond so i dunno how it does it and its a asm level of microoptimization